client/src/use/useStyling.ts

```diff
@@ -39,7 +39,7 @@ interface UpdateStylingArgs {
   fill?: boolean;
 }
 interface UseStylingParams {
-  markChangesPending: (name: string) => void;
+  markChangesPending: () => void;
```

Comment:
I don't know exactly how I feel about this and would just like to hear some arguments.  It's handy in the fact that useStyling doesn't care about the rest of the parameters for `markChangesPending` but it seems like it could make it confusing in terms of readability that all these locations are referencing the same exact function with different parameters.   It seems a bit like utilizing JS idea of an object and just modifying something during use.  I'm guessing this is dependent on how much we want to directly tie the `useSave.ts` implementation of `markChangesPending` to the styling.  If it is a direct connection that doesn't have other possibilities I suggest mimicking or hopefully finding a way to import the type?  I'm more interested to hear reasoning/discussion.

Reply:
In doing this, I intended there to be 2 "modes" of markChangesPending: changes about data that `useSave` should manage, and changes about data that some other module will manage.

`markChangesPending()` tells `useSave` that "I changed something, and you should show the user a pending save state, but don't worry about what I changed, I'll give you the data when the save button gets pressed".

`markChangesPending(type, track)` says "I changed this.  Hang onto it and send it to the server when I tell you".

I opted for this style mostly for compatibiltie's sake: to keep the diff small and disrupt less code.  If you want separate functions, for example

* `markChangesPending()`
* `markTrackChangePending(type, track)`

I think that would be a reasonable ask.  Your choice.

Reply:
I think it's okay to stay as is and the reasoning makes sense.  I was just putting myself into the shoes of someone who looked at this for the first time.